lib.php

```diff
@@ -36,36 +36,55 @@
 
 define('OBJECTFS_BYTES_IN_TERABYTE', 1099511627776);
 
-function update_object_record($contenthash, $location) {
-    global $DB;
 
-    $newobject = new \stdClass();
+/**
+ * @param string $contenthash
+ * @param int $newlocation
+ * @return stdClass
+ * @throws dml_exception
+ */
+function update_object_record($contenthash, $newlocation) {
```

Comment:
this is good but I found the name a little misleading, as I would have expected this to be the one which accepted an object. How about instead 'update_object_by_hash'

Reply:
updated to update_object_by_hash